dali/python/nvidia/dali/ops/_signatures.py

```diff
@@ -0,0 +1,432 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from inspect import Parameter, Signature
+import ast
+import os
+
+from pathlib import Path
+
+from typing import Union, Optional
+from typing import Sequence, List, Any
+
+from nvidia.dali import backend as _b
+from nvidia.dali import types as _types
+from nvidia.dali.ops import _registry, _names, _docs
+from nvidia.dali.fn import _to_snake_case
+from nvidia.dali import types
+from nvidia.dali import ops, fn
+
+
+def _create_annotation_placeholder(typename):
+    # Inspect and typing know better how to format annotations. They produce short representations
+    # of types that are builtins or belonging to typing module, but anything else gets a fully
+    # qualified path and there is no customization. Also the formatting using the `repr` of the type
+    # is driven by inspect.formatannotation (when we use the type directly) or by the internals of
+    # typing classes like Union. So we just pretend, that we are from typing, and we want to
+    # have our name as `typename`.
+
+    class _AnnotationPlaceholderMeta(type):
+        # We don't want <class 'typename'> when inspect calls inspect.formatannotation directly.
+        def __repr__(cls):
+            return typename
+
+    class _AnnotationPlaceholder(metaclass=_AnnotationPlaceholderMeta):
+        # typing uses those instead of repr
+        __name__ = typename
+        __qualname__ = typename
+        # only if you are part of `typing.` your path is hidden
+        __module__ = "typing"
```

Comment:
What do you mean by "path" here?

Reply:
fully qualified type name, so `module.submodule.class`.